app/src/main/java/org/walleth/activities/SwitchNetworkActivity.kt

```diff
@@ -48,7 +57,30 @@ open class SwitchNetworkActivity : AppCompatActivity() {
 
     private fun getAdapter() = NetworkAdapter(AllNetworkDefinitions) {
         networkDefinitionProvider.setCurrent(it)
-        finish()
+        if (GethLightEthereumService.shouldRun) {
```

Comment:
Unfortunately we cannot do it that way - the node has to listen on changes to the current network definition and restart then - otherwise we get in real trouble here: #99
Also if we would switch the network e.g. after scanning erc-681 with a different network-id this would not work - should listen on the effect and not one cause